lang/java/avro/src/main/java/org/apache/avro/file/DataFileStream.java

```diff
@@ -97,18 +97,30 @@ protected DataFileStream(DatumReader<D> reader) throws IOException {
     this.reader = reader;
   }
 
-  /** Initialize the stream by reading from its head. */
-  void initialize(InputStream in) throws IOException {
-    this.header = new Header();
-    this.vin = DecoderFactory.get().binaryDecoder(in, vin);
+  byte[] readMagic() throws IOException {
+    if (this.vin == null) {
+      throw new IOException("InputStream is not initialized");
+    }
     byte[] magic = new byte[DataFileConstants.MAGIC.length];
     try {
       vin.readFixed(magic); // read magic
     } catch (IOException e) {
       throw new IOException("Not an Avro data file.", e);
```

Comment:
always nice to include the error string of the nested exception, as there may be many other causes of problems than just file type

Reply:
+1. Didn't want to make additional changes to existing code to keep the changes minimal.

Reply:
Adding the exception cause already adds that message. We'll done.

Reply:
> Adding the exception cause adds that message. Let's do this the same as in line 108.

yes, if the whole exception history is printed. no if only the first error string gets logged, or the chain of causes gets stripped after a couple of wrappings (hello hive!). if the filename is known, that's handy to include too

Reply:
Filename isn't available. Null check has been added as a safeguard only.